contract/src/main/kotlin/io/provenance/scope/loan/contracts/AppendLoanDocumentsContract.kt

```diff
@@ -11,31 +11,36 @@ import io.provenance.scope.loan.LoanScopeFacts
 import io.provenance.scope.loan.utility.ContractRequirementType.VALID_INPUT
 import io.provenance.scope.loan.utility.documentValidation
 import io.provenance.scope.loan.utility.isValid
+import io.provenance.scope.loan.utility.orError
 import io.provenance.scope.loan.utility.validateRequirements
 import tech.figure.loan.v1beta1.LoanDocuments
 
-@Participants(roles = [PartyType.OWNER])
+@Participants(roles = [PartyType.OWNER]) // TODO: Should this eventually be changed to SERVICER?
 @ScopeSpecification(["tech.figure.loan"])
-open class AppendLoanDocContract(
+open class AppendLoanDocumentsContract(
     @Record(LoanScopeFacts.documents) val existingDocs: LoanDocuments,
 ) : P8eContract() {
 
-    @Function(invokedBy = PartyType.OWNER)
+    @Function(invokedBy = PartyType.OWNER) // TODO: Should this eventually be changed to SERVICER?
     @Record(LoanScopeFacts.documents)
     open fun appendDocuments(@Input(LoanScopeFacts.documents) newDocs: LoanDocuments): LoanDocuments {
         val newDocList = LoanDocuments.newBuilder().mergeFrom(existingDocs)
         validateRequirements(VALID_INPUT) {
+            /* Primitive type used for protobuf keys to avoid comparison interference from unknown fields */
             val existingDocChecksums = existingDocs.documentList.fold(mutableMapOf<String, Boolean>()) { acc, documentMetadata ->
                 acc.apply {
                     documentMetadata.checksum.takeIf { it.isValid() }?.checksum?.let { checksum ->
                         acc[checksum] = true
                     }
                 }
             }
+            requireThat(
+                newDocs.documentList.isNotEmpty() orError "Must supply at least one document"
+            )
             newDocs.documentList.forEach { doc ->
                 documentValidation(doc)
                 doc.checksum.checksum?.let { newDocChecksum ->
-                    if (existingDocChecksums[newDocChecksum] != true) { // TODO: Confirm that we want to silently ignore duplicates
+                    if (existingDocChecksums[newDocChecksum] != true) { // TODO: Confirm if duplicates silently ignored, or do like AppendLoanStates
```

Comment:
On this one, my stance would be if absolutely nothing changed (id, file_name, uri/checksum, etc.) then we can silently ignore because the outcome would be the same as is expected - that the file is in the list. Not as confusing as _not_ replacing the eNote.

Reply:
If the checksum is the same, but something else changed, we should have it fail with an appropriate message.

Reply:
Implemented in #20